src/Rpc/AuthInterceptor.cs

```diff
@@ -0,0 +1,36 @@
+using Grpc.Core;
+using Grpc.Core.Interceptors;
+using NodeGuard.Data.Repositories.Interfaces;
+
+namespace NodeGuard.Rpc;
+
+public class ServerAuthInterceptor : Interceptor
```

Comment:
What about `GRPCAuthInterceptor`? It would be nice also to have what's this class is about using the `///` comments way.

Reply:
Also the filename should match the class name per conventions.